src/taker/api.rs

```diff
@@ -1865,9 +1961,11 @@ impl Taker {
         network: Network,
         config: &TakerConfig,
     ) -> Result<(), TakerError> {
-        let offers =
-            sync_offerbook_with_addresses(get_advertised_maker_addresses(network).await?, config)
-                .await;
+        let offers = sync_offerbook_with_addresses(
+            get_advertised_maker_addresses(None, None, network).await?,
```

Comment:
This was is legacy code and as we are touching it, Lets break this into two distinct statements. First get all the advertised addresses from DNS. Then download offers. Would be clearer to read what the heck is going on. 

Also lets rename these two awkward names. How about

sync_offerbook_with_addresses -> fetch_offer_from_makers
get_advertised_maker_addresses -> fetch_addresses_from_dns